### DISCUSSION TITLE
Get a list of Data Objects from RootedSubgraphData Class

### DISCUSSION THREAD

**Discussion by MohSamNaf** [pyg-team/pytorch_geometric#8040] [Q&A]
Hi

I would like to use the transform classes such as `RootedRWSubgraph `and `RootedEgoNets`.

I was able to use them as they produce 5 tensors. The tensors corresponds to the following:

>     

    * :obj:`sub_edge_index` (Tensor): The edge indices of all combined rooted
      subgraphs.
    * :obj:`n_id` (Tensor): The indices of nodes in all combined rooted
      subgraphs.
    * :obj:`e_id` (Tensor): The indices of edges in all combined rooted
      subgraphs.
    * :obj:`n_sub_batch` (Tensor): The batch vector to distinguish nodes across
      different subgraphs.
    * :obj:`e_sub_batch` (Tensor): The batch vector to distinguish edges across
      different subgraphs.

An example that I was using:
```
from torch_geometric.data import Data
from torch_geometric.transforms import RootedRWSubgraph

def generate_rooted_rw_subgraphs(graph, walk_length, repeat=1):
    """
    Generate rooted random walk subgraphs rooted at every node in the input graph.

    Args:
        graph (Data): The input graph.
        walk_length (int): The length of the random walk.
        repeat (int, optional): The number of times to repeat the random walk.
            (default: 1)

    Returns:
        List[Tuple]: A list of tuples, each representing a rooted random walk subgraph represented in 5 tensors.
    """
    subgraphs = []

    # Create a RootedRWSubgraph object
    rw_subgraph = RootedRWSubgraph(walk_length, repeat)

    # Extract the rooted random walk subgraph for the current node
    subgraph_data = rw_subgraph.extract(graph)
    

    return subgraph_data

if __name__ == "__main__":
    import torch
    from torch_geometric.data import Data

    # Define the edge connections for the graph to ensure that all nodes have at least one edge.
    edge_index = torch.tensor([[0, 1, 1, 2, 2, 3, 4, 0],  # Edges
                            [1, 0, 2, 1, 3, 2, 0, 4]], dtype=torch.long)

    # Node features (5 nodes with 3 features each)
    x = torch.tensor([[1.0, 2.0, 3.0],
                    [4.0, 5.0, 6.0],
                    [7.0, 8.0, 9.0],
                    [10.0, 11.0, 12.0],
                    [13.0, 14.0, 15.0]], dtype=torch.float)

    # Create the Data object for the graph
    graph = Data(x=x, edge_index=edge_index)
    
    print("Input Graph: ", graph)

    # Generate rooted random walk subgraphs
    walk_length = 3
    repeat = 1
    subgraph_data = generate_rooted_rw_subgraphs(graph, walk_length, repeat)
    
    print("Rooted Random Walk Subgraphs: ", subgraph_data)

```
 
The main question is I would like to know how can I make use of these outputs to reconstruct a `Data` object that corresponds to each homogeneous  subgraph? 

Any help is appreciated. 
Thank you!

**Comment by rusty1s** [comment]
You can call `subgraph_data.map_data()` to reconstruct a `Data` object.

**Comment by MohSamNaf** [comment]
Thank you for your reply.

It gives me the error `AttributeError: 'tuple' object has no attribute 'map_data'` The 5 tensors are grouped in a tuple consisting of 5 tensors

**Reply by rusty1s** [reply]
You need to call via `subgraph_data = rw_subgraph(graph)` instead of `subgraph_data = rw_subgraph.extract(graph)`.
